react/features/calendar-sync/middleware.js

```diff
@@ -265,24 +269,39 @@ function _updateCalendarEntries(events) {
     if (events && events.length) {
         // eslint-disable-next-line no-invalid-this
         const { dispatch, getState } = this;
-
         const knownDomains = getState()['features/base/known-domains'];
-        const eventList = [];
-
         const now = Date.now();
+        const entryMap = new Map();
 
         for (const event of events) {
             const calendarEntry = _parseCalendarEntry(event, knownDomains);
 
-            calendarEntry
-                && calendarEntry.endDate > now
-                && eventList.push(calendarEntry);
+            if (calendarEntry && calendarEntry.endDate > now) {
+                // This is the data structure we'll try to match all entries to.
+                // The smaller the better, hence the single letter field names.
+                const eventMatchHash = md5.hex(JSON.stringify({
+                    d: new Date(calendarEntry.startDate).toTimeString(),
+                    t: calendarEntry.title,
+                    u: calendarEntry.url
+                }));
+                const existingEntry = entryMap.get(eventMatchHash);
+
+                // We need to dedupe the list based on title, URL and time of
+                // the day, and only show the first occurence. So if there was
+                // no matcing entry or there was, but its a later event, we
+                // overwrite/save it in the map.
+                if (!existingEntry
+                    || existingEntry.startDate > event.startDate) {
+                    entryMap.set(eventMatchHash, calendarEntry);
+                }
+            }
         }
 
         dispatch(
             setCalendarEvents(
-                eventList
+                Array.from(entryMap.values())
                     .sort((a, b) => a.startDate - b.startDate)
-                    .slice(0, MAX_LIST_LENGTH)));
+                    .slice(0, MAX_LIST_LENGTH)
+            ));
```

Comment:
We don't close functions like that! Please move the closing at the end of the line above.

We do have cases in which closing appears on a new line like you've written it but actually they're closing other things as well such as objects so the closing of the function still happens at the end of the line, not on a new line:

````javascript
f({
    prop1: value1,
    prop2: value2
});
````

Reply:
Fixed, thanks!

Btw, isn't there a formatting option to force this?